numpy/core/src/multiarray/common.h

```diff
@@ -169,6 +179,11 @@ check_and_adjust_axis(int *axis, int ndim)
     }
     return 0;
 }
+static NPY_INLINE int
+check_and_adjust_axis(int *axis, int ndim)
+{
+    return check_and_adjust_axis_msg(axis, ndim, Py_None);
```

Comment:
ok, this goes a bit off-topic. I [still think](https://github.com/numpy/numpy/pull/8843#discussion_r108532657) the code is correct wrt reference counting.